src/x/net/http/convert.go

```diff
@@ -38,6 +39,77 @@ var (
 	errDurationType = errors.New("invalid duration type")
 )
 
+// NanosToDurationBytes transforms a json byte slice with Nano keys into Duration keys.
+func NanosToDurationBytes(r io.Reader) (map[string]interface{}, error) {
+	var dict map[string]interface{}
+	d := json.NewDecoder(r)
+	d.UseNumber()
```

Comment:
You're already handling the `float64` case below, wouldn't it be simpler just to stick with that?

Reply:
So I think we actually want something like this `{"fieldNanos":100.5}` to fail (e.g. nanos shouldn't be floats). I'm fine with removing the `case float64` below though as it will never actually be called.